### PR TITLE
[Tests-Only] Refactor apiShareCreateSpecial1 to use Shares folder

### PR DESCRIPTION
**Pull request by haribhandari07**
## Description

This PR makes `apiShareCreateSpecial1` test scenarios to be valid in both oC10 and OCIS. OCIS always puts received shares into a folder called Shares for the share receiver. On OCIS, auto-accept shares is always disabled.

## Related Issue

- Part of https://github.com/owncloud/ocis/issues/518
## How Has This Been Tested?

- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature [hunk 1]
```diff
@@ -0,0 +1,760 @@
...
+      | file_target | /Shares/Shares/FOLDER |
+      | uid_owner   | %username%            |
+      | expiration  | +15 days              |
+      | share_with  | %username%            |
+    And the response when user "Brian" gets the info of the last share should include
+      | expiration | +15 days |
+    Examples:
+      | ocs_api_version | ocs_status_code | http_status_code |
+      | 1               | 100             | 200              |
+      | 2               | 200             | 200              |
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature:52]: The `/Shares/Shares/FOLDER` value is issue #37883 
Tag this scenario `notToImplementOnOCIS` and `issue-37883`
Make another similar scenario with the correct expected value `/Shares/FOLDER` and tag it `skipOnOcV10`
That will make sure that we only try to implement code for "good" scenarios in OCIS.

File: tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature [hunk 2]
```diff
@@ -0,0 +1,760 @@
...
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature:78]: The `/Shares/Shares/FOLDER` value is issue #37883 
Tag this scenario `notToImplementOnOCIS` and `issue-37883`
Make another similar scenario with the correct expected value `/Shares/FOLDER` and tag it `skipOnOcV10`
That will make sure that we only try to implement code for "good" scenarios in OCIS.

File: tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature [hunk 3]
```diff
@@ -0,0 +1,760 @@
...
+      | file_target | /Shares/Shares/FOLDER |
+      | uid_owner   | %username%            |
+      | expiration  | +15 days              |
+      | share_with  | grp1                  |
+    And the response when user "Brian" gets the info of the last share should include
+      | expiration | +15 days |
+    Examples:
+      | ocs_api_version | ocs_status_code | http_status_code |
+      | 1               | 100             | 200              |
+      | 2               | 200             | 200              |
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature:181]: Another one that demonstrates issue #37883

File: tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature [hunk 4]
```diff
@@ -0,0 +1,760 @@
...
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature:209]: Another one that demonstrates issue #37883

File: tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature [hunk 5]
```diff
@@ -0,0 +1,760 @@
...
+      | file_target | /Shares/Shares/textfile0.txt |
+      | uid_owner   | %username%                   |
+      | share_with  | %username%                   |
+      | expiration  | +7 days                      |
+    And the response when user "Brian" gets the info of the last share should include
+      | expiration | +7 days |
+    Examples:
+      | ocs_api_version |
+      | 1               |
+      | 2               |
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature:289]: Another one that demonstrates issue #37883

File: tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature [hunk 6]
```diff
@@ -0,0 +1,760 @@
...
+      | file_target | /Shares/Shares/textfile0.txt |
+      | uid_owner   | %username%                   |
+      | share_with  | %username%                   |
+      | expiration  | +30 days                     |
+    And the response when user "Brian" gets the info of the last share should include
+      | expiration | +30 days |
+    Examples:
+      | ocs_api_version |
+      | 1               |
+      | 2               |
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature:332]: Another one that demonstrates issue #37883

File: tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature [hunk 7]
```diff
@@ -0,0 +1,760 @@
...
+      | file_target | /Shares/Shares/textfile0.txt |
+      | uid_owner   | %username%                   |
+      | share_with  | grp1                         |
+      | expiration  | +7 days                      |
+    And the response when user "Brian" gets the info of the last share should include
+      | expiration | +7 days |
+    Examples:
+      | ocs_api_version |
+      | 1               |
+      | 2               |
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature:413]: Another one that demonstrates issue #37883

File: tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature [hunk 8]
```diff
@@ -0,0 +1,760 @@
...
+      | file_target | /Shares/Shares/textfile0.txt |
+      | uid_owner   | %username%                   |
+      | share_with  | grp1                         |
+      | expiration  | +30 days                     |
+    And the response when user "Brian" gets the info of the last share should include
+      | expiration | +30 days |
+    Examples:
+      | ocs_api_version |
+      | 1               |
+      | 2               |
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature:460]: Another one that demonstrates issue #37883

**Review by phil-davis — CHANGES_REQUESTED** [review]

**Comment by phil-davis** [comment]
I also raised issue #37944 - it applies to a lot of these "sharing to share_folder" scenarios in multiple PRs. So we can sort that out next week.

File: tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature [hunk 9]
```diff
@@ -0,0 +1,760 @@
...
+
+  @skipOnOcV10.3
+  Scenario Outline: sharing with default expiration date enabled but not enforced for users, user shares without specifying expireDate
+    Given using OCS API version "<ocs_api_version>"
+    And parameter "shareapi_default_expire_date_user_share" of app "core" has been set to "yes"
+    And user "Brian" has been created with default attributes and without skeleton files
+    When user "Alice" shares folder "/FOLDER" with user "Brian" using the sharing API
+    And user "Brian" accepts share "/FOLDER" offered by user "Alice" using the sharing API
+    Then the OCS status code should be "<ocs_status_code>"
+    And the HTTP status code should be "<http_status_code>"
```
Comment (dpakach) [inline tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature:17]: This step is supposed to check the status code of sharing step but here it checks the status of accepting the share.

File: tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature [hunk 10]
```diff
@@ -0,0 +1,760 @@
...
+    And user "Brian" has been created with default attributes and without skeleton files
+    When user "Alice" creates a share using the sharing API with settings
+      | path        | /FOLDER    |
+      | shareType   | user       |
+      | shareWith   | Brian      |
+      | permissions | read,share |
+      | expireDate  | +15 days   |
+    And user "Brian" accepts share "/FOLDER" offered by user "Alice" using the sharing API
+    Then the OCS status code should be "<ocs_status_code>"
+    And the HTTP status code should be "<http_status_code>"
```
Comment (dpakach) [inline tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature:40]: This step is supposed to check the status code of sharing step but here it checks the status of accepting the share.

File: tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature [hunk 11]
```diff
@@ -0,0 +1,760 @@
...
```
Comment (dpakach) [inline tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature:66]: This step is supposed to check the status code of sharing step but here it checks the status of accepting the share.

File: tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature [hunk 12]
```diff
@@ -0,0 +1,760 @@
...
+  Scenario Outline: sharing with default expiration date enabled but not enforced for groups, user shares without specifying expireDate
+    Given using OCS API version "<ocs_api_version>"
+    And parameter "shareapi_default_expire_date_group_share" of app "core" has been set to "yes"
+    And user "Brian" has been created with default attributes and without skeleton files
+    And group "grp1" has been created
+    And user "Brian" has been added to group "grp1"
+    When user "Alice" shares folder "/FOLDER" with group "grp1" using the sharing API
+    And user "Brian" accepts share "/FOLDER" offered by user "Alice" using the sharing API
+    Then the OCS status code should be "<ocs_status_code>"
+    And the HTTP status code should be "<http_status_code>"
```
Comment (dpakach) [inline tests/acceptance/features/apiShareCreateSpecialShares1/createShareExpirationDate.feature:144]: This step is supposed to check the status code of sharing step but here it checks the status of accepting the share.

File: tests/acceptance/features/apiShareCreateSpecialShares1/createShareReceivedInMultipleWays.feature [hunk 13]
```diff
@@ -0,0 +1,192 @@
...
+    Given using OCS API version "<ocs_api_version>"
+    And user "Brian" has been created with default attributes and without skeleton files
+    And group "grp1" has been created
+    And user "Brian" has been added to group "grp1"
+    And user "Alice" has shared file "welcome.txt" with group "grp1"
+    And user "Brian" has accepted share "/welcome.txt" offered by user "Alice"
+    When user "Alice" shares file "/welcome.txt" with user "Brian" using the sharing API
+    And user "Brian" accepts share "/welcome.txt" offered by user "Alice" using the sharing API
+    Then the OCS status code should be "<ocs_status_code>"
+    And the HTTP status code should be "200"
```
Comment (dpakach) [inline tests/acceptance/features/apiShareCreateSpecialShares1/createShareReceivedInMultipleWays.feature:19]: This step is supposed to check the status code of sharing step but here it checks the status of accepting the share.

File: tests/acceptance/features/apiShareCreateSpecialShares1/createShareReceivedInMultipleWays.feature [hunk 14]
```diff
@@ -0,0 +1,192 @@
...
+    Given using OCS API version "<ocs_api_version>"
+    And user "Brian" has been created with default attributes and without skeleton files
+    And group "grp1" has been created
+    And user "Alice" has created folder "/test"
+    And user "Alice" has created folder "/test/sub"
+    And user "Alice" has shared folder "/test" with group "grp1"
+    When user "Alice" shares folder "/test/sub" with user "Brian" using the sharing API
+    And user "Brian" accepts share "/test/sub" offered by user "Alice" using the sharing API
+    Then the OCS status code should be "<ocs_status_code>"
+    And the HTTP status code should be "200"
```
Comment (dpakach) [inline tests/acceptance/features/apiShareCreateSpecialShares1/createShareReceivedInMultipleWays.feature:73]: This step is supposed to check the status code of sharing step but here it checks the status of accepting the share.

**Review by dpakach — DISMISSED** [review]
Some scenarios have 
`Then the status code should be <code>` step which checks the status of accepting the share not creating the share. 
I have not commented on all of them. Please go through all the scenarios fix those to check the status of Sharing step.

**Review by phil-davis — APPROVED** [review]

**Pull request merged**